source/github-helpers/get-default-branch.ts

```diff
@@ -13,15 +13,9 @@ import {getRepo, getCurrentBranchFromFeed} from '.';
 // "This branch is 1 commit ahead, 27 commits behind master."
 const branchInfoRegex = /([^ ]+)\.$/;
 
-const getDefaultBranch = cache.function(async function (repository?: pageDetect.RepositoryInfo): Promise<string> {
-	if (arguments.length === 0) {
-		repository = getRepo();
-	}
-
-	if (!repository) {
-		throw new Error('getDefaultBranch was called on a non-repository page');
-	}
-
+// DO NOT use optional arguments/defaults in "cached functions" because they can't be memoized effectively
+// https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1864
```

Comment:
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1864 is only an issue because `cacheKey` just isn't followed internally.